rmgpy/data/surface.py

```diff
@@ -0,0 +1,300 @@
+                                                                  entry.surface_site_density.units))
+    if entry.facet:
+        f.write('    facet = "{0}",\n'.format(entry.facet))
+    if entry.metal:
+        f.write('    metal = "{0}",\n'.format(entry.metal))
+    f.write(f'    shortDesc = """{entry.short_desc.strip()}""",\n')
+    f.write(f'    longDesc = \n"""\n{entry.long_desc.strip()}\n""",\n')
+    f.write(')\n\n')
+
+
+################################################################################
+
+
+################################################################################
+
+class MetalLibrary(Database):
+    """
+    A class for working with a RMG metal library.
+    """
+
+    def __init__(self, label='', name='', short_desc='', long_desc=''):
+        Database.__init__(self, label=label, name=name, short_desc=short_desc, long_desc=long_desc)
+
+    def load_entry(self,
+                   index,
+                   label,
+                   metal='',
+                   facet='',
+                   surfaceSiteDensity=None,
+                   bindingEnergies=None,
+                   shortDesc='',
+                   longDesc='',
+                   ):
+        """
+        Method for parsing entries in database files.
+        Note that these argument names are retained for backward compatibility.
+        """
+        if bindingEnergies:
+            binding_energies = dict()
+            for element, energy in bindingEnergies.items():
+                binding_energies[element] = rmgpy.quantity.Energy(energy)
+        else:
+            binding_energies = None
+        
+        if surfaceSiteDensity:
+            surface_site_density = rmgpy.quantity.SurfaceConcentration(*surfaceSiteDensity)
+        else:
+            surface_site_density = None
+
+        self.entries[label] = Entry(
+            index=index,
+            label=label,
+            metal=metal,
+            facet=facet,
+            surface_site_density=surface_site_density,
+            binding_energies=binding_energies,
+            short_desc=shortDesc,
+            long_desc=longDesc.strip(),
+        )
+
+    def load(self, path):
+        """
+        Load the metal library from the given path
+        """
+        Database.load(self, path, local_context={}, global_context={})
+
+    def save_entry(self, path, entry):
+        """
+        Write the given `entry` in the metal database to the file object `f`.
+        """
+        return save_entry(path, entry)
+
+    def get_binding_energies(self, label):
+        """
+        Get a metal's binding energies from its label.
+
+        Raises DatabaseError (rather than returning None) if it can't be found.
+        """
+        try:
+            binding_energies = self.entries[label].binding_energies
+        except KeyError:
+            raise DatabaseError(f'Metal {label!r} not found in metal library database.')
```

Comment:
This is the error I get when specifying metal="Ni". Would it make sense to include a reminder on how to format the label in this error message?

Reply:
yep, that's a great suggestion!